### PR TITLE
silence -Wstring-truncation for correct use of strncpy

### PR DESCRIPTION
**Pull request by nmschulte**
- ensuring header is zerod

addresses #5145

### CODE REVIEW

**Comment by rusefillc** [comment]
I think it's null terminated or max length? I guess based on lack of length

File: firmware/console/binary_log/log_field.cpp [hunk 1]
```diff
@@ -43,10 +43,9 @@ void LogField::writeHeader(Writer& outBuffer) const {
 	buffer[54] = m_digits;
 
 	// Offset 55, (optional) category string
+	memset(&buffer[55], 0, 34);
```
Comment (mck1117) [inline firmware/console/binary_log/log_field.cpp:46]: we have `efi::clear`, maybe useful

**Comment by nmschulte** [comment]
It's unclear to me why `stringop-truncated` is not treated as an error; there is some interplay between the `RUSEFI_OPT` values of `firmware/rusefi_rules.mk` (fed to `USE_OPTS` of ChibiOS IIUC) and `firmware/Makefile`'s `CWARN`/`CPPWARN`, and/or GCC behavior re: order of flags that I do not understand.  I saw also some `maybe-unitialized` warns/errors (from the linker) in my confusing trials that may be going unnoticed.

_(Does `USE_OPTS` only apply to ChibiOS code, leading to my confusion?  Is `-Werror` and `-Werror=all` the same to GCC?  Should strive for `CWARN`/`CPPWARN` to have `-Werror`[`=all`]?)_

File: firmware/console/binary_log/log_field.cpp (hunk 1 above)
Comment (nmschulte) [reply firmware/console/binary_log/log_field.cpp:46]: I poked at this but the generics confused me: does this support `char[]` yet?

**Pull request merged**